tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs

```diff
@@ -2136,5 +2153,72 @@ public void TestApply()
                 Assert.Equal(newCol[i], values[i] + 0.5d);
             }
         }
+
+        [Fact]
+        public void TestRollingColumnWindow()
+        {
+            int?[] values = { 1, 2, 3, 4, 5, 4, null, null, -1 };
+            PrimitiveDataFrameColumn<int> ints = new PrimitiveDataFrameColumn<int>("Ints", values);
+            PrimitiveDataFrameColumn<int> sum = ints.Rolling(2).Apply((LinkedList<int?> windowValues, long currentIndex) =>
+            {
+                if (windowValues.Count < 2)
+                {
+                    return null;
+                }
+                return windowValues.Sum();
+            });
+
+            int?[] verifySum = { null, 3, 5, 7, 9, 9, 4, null, -1 };
+            PrimitiveDataFrameColumn<int> verifySumColumn = new PrimitiveDataFrameColumn<int>("Verify", verifySum);
+            Assert.True(verifySumColumn.ElementwiseEquals(sum).All());
```

Comment:
I wonder if we shouldn't have a helper method that does this comparison, and that helper method would be able to spit out detailed failure information about which values didn't match, at what index, etc.

The way it is written now, if a failure happens, all the diagnostic information you get is "Expected True, got False".

Reply:
I'll set this up tomorrow. I think the rest of the comments are addressed.